qos: T1871: add MTU option when configure limiter traffic-policy

HollyGurza opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Text placed between these delimiters is considered a comment and is not rendered -->

## Change Summary
<!--- Provide a general summary of your changes in the Title above -->
Added MTU option for qos limiter traffic policy
## Types of changes
<!---
What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
the box, please use [x]
-->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- All submitted PRs must be linked to a Task on Phabricator. -->
* https://vyos.dev/T1871

## Related PR(s)
<!-- Link here any PRs in other repositories that are required by this PR -->

## Component(s) name
<!-- A rather incomplete list of components: ethernet, wireguard, bgp, mpls, ldp, l2tp, dhcp ... -->

## Proposed changes
<!--- Describe your changes in detail -->

## How to test
<!---
Please describe in detail how you tested your changes. Include details of your testing
environment, and the tests you ran. When pasting configs, logs, shell output, backtraces,
and other large chunks of text, surround this text with triple backtics
```
like this
```
-->
```
set qos policy limiter T1871 default bandwidth '10mbit'
set qos policy limiter T1871 default burst '100kb'
set qos policy limiter T1871 default mtu 10000
set qos interface eth0 ingress T1871 
```

## Smoketest result
<!-- Provide the output of the smoketest
-->
```
vyos@vyos:~$ python3 /usr/libexec/vyos/tests/smoke/cli/test_qos.py 
test_01_cake (__main__.TestQoS.test_01_cake) ... ok
test_02_drop_tail (__main__.TestQoS.test_02_drop_tail) ... ok
test_03_fair_queue (__main__.TestQoS.test_03_fair_queue) ... ok
test_04_fq_codel (__main__.TestQoS.test_04_fq_codel) ... ok
test_05_limiter (__main__.TestQoS.test_05_limiter) ... ok
test_06_network_emulator (__main__.TestQoS.test_06_network_emulator) ... ok
test_07_priority_queue (__main__.TestQoS.test_07_priority_queue) ... ok
test_08_random_detect (__main__.TestQoS.test_08_random_detect) ... skipped 'tc returns invalid JSON here - needs iproute2 fix'
test_09_rate_control (__main__.TestQoS.test_09_rate_control) ... ok
test_10_round_robin (__main__.TestQoS.test_10_round_robin) ... ok
test_11_shaper (__main__.TestQoS.test_11_shaper) ... ok

----------------------------------------------------------------------
Ran 11 tests in 40.376s

OK (skipped=1)

```

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

HollyGurza committed:
qos: T1871: add MTU option when configure limiter traffic-policy

add mtu to default and specified class
update smoke test

dmbaturin commented on interface-definitions/include/qos/mtu.xml.i:
```diff
@@ -0,0 +1,15 @@
+    </valueHelp>
+    <valueHelp>
+      <format>&lt;number&gt;&lt;suffix&gt;</format>
+      <description>Bytes with scaling suffix (kb, mb, gb)</description>
```
Does it actually accept gigabyte-large MTUs? Even if yes — should we allow that?

c-po replied:
> Does it actually accept gigabyte-large MTUs? Even if yes — should we allow that?

I'm unaware of a Gigabyte MTU 

Please use the common MTU definition instead: https://github.com/vyos/vyos-1x/blob/current/interface-definitions/include/interface/mtu-68-16000.xml.i

HollyGurza replied:
68 is too low for tc possible value is 256 - 4294967295
```
vyos@vyos:~$  tc filter replace dev eth0 parent ffff: prio 255 protocol all basic action police conform-exceed drop/ok rate 10000000 burst 100kb mtu 68
RTNETLINK answers: Cannot allocate memory
We have an error talking to the kernel
vyos@vyos:~$  tc filter replace dev eth0 parent ffff: prio 255 protocol all basic action police conform-exceed drop/ok rate 10000000 burst 100kb mtu 255
RTNETLINK answers: Cannot allocate memory
We have an error talking to the kernel
vyos@vyos:~$  tc filter replace dev eth0 parent ffff: prio 255 protocol all basic action police conform-exceed drop/ok rate 10000000 burst 100kb mtu 256
vyos@vyos:~$  tc filter replace dev eth0 parent ffff: prio 255 protocol all basic action police conform-exceed drop/ok rate 10000000 burst 100kb mtu 4294967295
vyos@vyos:~$  tc filter replace dev eth0 parent ffff: prio 255 protocol all basic action police conform-exceed drop/ok rate 10000000 burst 100kb mtu 4294967296
Error: argument "4294967296" is wrong: mtu
```

in the task description, author said `the packet size between ip stack and nic could be 65536` therefore range 68-16000 probably is incorrect for this case, I think we can use range 256-65535.

c-po replied:
Then please limit to this range

HollyGurza replied:
done

c-po reviewed (changes requested):
Please also add the missing output of the smoketests and extend them with this new MTU option so it's always verified on image build.

https://github.com/vyos/vyos-1x/blob/current/smoketest/scripts/cli/test_qos.py

HollyGurza commented:
> Please also add the missing output of the smoketests and extend them with this new MTU option so it's always verified on image build.
> 
> https://github.com/vyos/vyos-1x/blob/current/smoketest/scripts/cli/test_qos.py

done, added mtu and other missing options for limiter test.
also added mtu to `class` section, in the previous iteration it was only in the default section

<img width="737" alt="Screenshot 2024-03-15 at 7 27 44 PM" src="https://github.com/vyos/vyos-1x/assets/1856505/fcd2a666-b8a8-4c2d-a4f4-9c6350fc3945">

c-po commented on smoketest/scripts/cli/test_qos.py:
```diff
@@ -279,6 +291,15 @@ def test_05_limiter(self):
+                    self.cli_set(qos_class_base + ['burst', qos_class_config['burst']])
+
+                if 'mtu' in qos_class_config:
+                    self.cli_set(qos_class_base + ['mtu', qos_class_config['mtu']])
```
I see the MTU programmed to the kernel, but there is no validation who checks if the programmed value is the value when it's read back.

HollyGurza replied:
done. Added validate what exactly is set to tc policy, not ideal but works.

HollyGurza commented:
@c-po Need backport to Sagitta?

<img width="248" alt="Screenshot 2024-03-20 at 1 05 19 PM" src="https://github.com/vyos/vyos-1x/assets/1856505/49a1bb1a-3144-454e-9400-f2cf2df90f14">

c-po commented:
@MergifyIo backport sagitta
